src/ILCompiler.TypeSystem/tests/VirtualMethodEnumerationAlgorithmTests.cs

```diff
@@ -64,5 +64,20 @@ public void TestCachingEnumerationAlgorithm()
             Assert.Equal("ToString", testTypeMethods[0].Name);
             Assert.Equal(_testType, testTypeMethods[0].OwningType);
         }
+
+        [Fact]
+        public void TestFindBaseUnificationGroup()
```

Comment:
@nattress Once you move this block to the other file too, the tests LGTM.

The functionality change should still be reviewed by @davidwrighton. I lack expertise in that part of the codebase.